### PR TITLE
Update RBAC API versions to avoid deprecations in CNI helm chart

### PR DESCRIPTION
**Pull request by glitchcrab**
When testing the `linkerd2-cni` chart with `ct`, it flags up usage
of some deprecated apiVersions.

This PR aligns the RBAC API group across all resources in the chart.

---

Signed-off-by: Simon Weald <glitchcrab-github@simonweald.com>

### CODE REVIEW

**Comment by adleong** [comment]
@zaharidichev can you take a look at this?

**Review by Pothulapati — APPROVED** [review]
LGTM! Thanks for the PR

Looks like we already use `rbac.authorization.k8s.io/v1` [everywhere](https://github.com/linkerd/linkerd2/blob/main/charts/linkerd2/templates/web-rbac.yaml#L6), so this should be a no brainer!

**Review by kleimkuhler — APPROVED** [review]
@glitchcrab Could you please provide the `ct` command you are running to see these lints? I was unfamiliar with this tool and have it downloaded now, but I may be missing a config for `chart_schema.yaml` or something.

That being said, this looks good given that the CNI integration test still passes and we use this version value everywhere else.

**Comment by glitchcrab** [comment]
@kleimkuhler sure, I believe it's just the standard config taken from the helmpack/chart-testing image:

```
/ # cat /etc/ct/chart_schema.yaml
name: str()
home: str()
version: str()
apiVersion: str()
appVersion: any(str(), num())
description: str()
keywords: list(str(), required=False)
sources: list(str(), required=False)
maintainers: list(include('maintainer'), required=False)
icon: str(required=False)
engine: str(required=False)
condition: str(required=False)
tags: str(required=False)
deprecated: bool(required=False)
kubeVersion: str(required=False)
annotations: map(str(), str(), required=False)
---
maintainer:
  name: str()
  email: str(required=False)
  url: str(required=False)
```

We do also pass the cmdline flag `validate-maintainers=false`.

**Comment by kleimkuhler** [comment]
@glitchcrab Thanks! Made it a little further but still have issues using the tool. I won't block on this, but curious if you've seen this or if you have the exact command you use?

It fails for me with `Error processing charts` and I'm not seeing anything obvious to fix it in the project issues.

```
❯ ct lint --validate-maintainers=false charts/linkerd2-cni/
Linting charts...
Directory 'charts/add-ons/grafana' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/grafana' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/grafana/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/grafana/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/grafana' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/prometheus' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/tracing' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/tracing/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/tracing/templates' is not a valid chart directory. Skipping...
Directory 'charts/add-ons/tracing' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster/templates' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster/templates' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster/templates' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster/templates' is not a valid chart directory. Skipping...
Directory 'charts/linkerd2-multicluster' is not a valid chart directory. Skipping...

------------------------------------------------------------------------------------------------------------------------
 Charts to be processed:
------------------------------------------------------------------------------------------------------------------------
 linkerd2-cni => (version: "0.1.0", path: "charts/linkerd2-cni")
 linkerd2 => (version: "0.1.0", path: "charts/linkerd2")
 partials => (version: "0.1.0", path: "charts/partials")
 patch => (version: "0.1.0", path: "charts/patch")
------------------------------------------------------------------------------------------------------------------------

Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "linkerd2" chart repository
...Successfully got an update from the "linkerd" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Deleting outdated charts
Linting chart 'linkerd2-cni => (version: "0.1.0", path: "charts/linkerd2-cni")'
Checking chart 'linkerd2-cni => (version: "0.1.0", path: "charts/linkerd2-cni")' for a version bump...
Old chart version: 0.1.0
New chart version: 0.1.0
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "linkerd2" chart repository
...Successfully got an update from the "linkerd" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 3 charts
Deleting outdated charts
Linting chart 'linkerd2 => (version: "0.1.0", path: "charts/linkerd2")'
Checking chart 'linkerd2 => (version: "0.1.0", path: "charts/linkerd2")' for a version bump...
Old chart version: 0.1.0
New chart version: 0.1.0
Linting chart 'partials => (version: "0.1.0", path: "charts/partials")'
Checking chart 'partials => (version: "0.1.0", path: "charts/partials")' for a version bump...
Old chart version: 0.1.0
New chart version: 0.1.0
Hang tight while we grab the latest from your chart repositories...
...Successfully got an update from the "linkerd2" chart repository
...Successfully got an update from the "linkerd" chart repository
Update Complete. ⎈Happy Helming!⎈
Saving 1 charts
Deleting outdated charts
Linting chart 'patch => (version: "0.1.0", path: "charts/patch")'
Checking chart 'patch => (version: "0.1.0", path: "charts/patch")' for a version bump...
Old chart version: 0.1.0
New chart version: 0.1.0
------------------------------------------------------------------------------------------------------------------------
 ✖︎ linkerd2-cni => (version: "0.1.0", path: "charts/linkerd2-cni") > Chart version not ok. Needs a version bump!
 ✖︎ linkerd2 => (version: "0.1.0", path: "charts/linkerd2") > Chart version not ok. Needs a version bump!
 ✖︎ partials => (version: "0.1.0", path: "charts/partials") > Chart version not ok. Needs a version bump!
 ✖︎ patch => (version: "0.1.0", path: "charts/patch") > Chart version not ok. Needs a version bump!
------------------------------------------------------------------------------------------------------------------------
Error: Error linting charts: Error processing charts
Error linting charts: Error processing charts
```

**Review by zaharidichev — APPROVED** [review]
LGTM and thank  you

**Pull request merged**
